demo/admin/src/products/ManufacturerFilter.tsx

```diff
@@ -0,0 +1,72 @@
+import { gql, useQuery } from "@apollo/client";
+import Autocomplete from "@mui/material/Autocomplete";
+import TextField from "@mui/material/TextField";
+import { GridFilterInputValueProps, GridFilterOperator } from "@mui/x-data-grid-pro";
+import * as React from "react";
+import { useIntl } from "react-intl";
+
+import { GQLManufacturersFilterQuery, GQLManufacturersFilterQueryVariables } from "./ManufacturerFilter.generated";
+
+const manufacturersQuery = gql`
+    query ManufacturersFilter($offset: Int!, $limit: Int!, $search: String) {
+        manufacturers(offset: $offset, limit: $limit, search: $search) {
+            nodes {
+                id
+                name
+            }
+            totalCount
+        }
+    }
+`;
+
+// Source: https://mui.com/x/react-data-grid/filtering/customization/#multiple-values-operator
+function ManufacturerFilter({ item, applyValue }: GridFilterInputValueProps) {
+    const intl = useIntl();
+
+    const { data } = useQuery<GQLManufacturersFilterQuery, GQLManufacturersFilterQueryVariables>(manufacturersQuery, {
+        variables: {
+            offset: 0,
+            limit: 10,
+        },
+    });
+
+    // source https://mui.com/material-ui/react-autocomplete/
+    return (
+        <Autocomplete
+            size="small"
+            options={data?.manufacturers.nodes ?? []}
+            autoHighlight
+            value={item.value}
+            isOptionEqualToValue={(option, value) => {
+                // does only highlight the selected value in options-list but does not trigger getOptionLabel-Call
+                return option.id == value;
+            }}
+            getOptionLabel={(option) => {
+                // option.name is not set if loaded from filters-model, small delay replacing id with name because of loading
+                return option.name ?? data?.manufacturers.nodes.find((item) => item.id === option)?.name ?? option;
+            }}
+            onChange={(event, value, reason) => {
+                // value can't be "{ id: value.id, name: value.name }" because value is sent to api
```

Comment:
Should we change this before sending it to the API? Would this help?

Reply:
if there was the possibility to handle/convert this in this filter-component, then yes :-)